src/Common/Misc/AnnotationBuilder/index.js

```diff
@@ -67,6 +67,28 @@ function fork(annotationObj) {
   return Object.assign({}, annotationObj, { generation, id });
 }
 
+function setDefaultName(annotationObject) {
+  if (annotationObject.selection.type === 'range') {
+    const rangeNames = Object.keys(annotationObject.selection.range.variables);
+    if (rangeNames.length > 0) {
+      annotationObject.name = rangeNames[0];
+      if (rangeNames.length > 1) {
+        annotationObject.name += ` & ${rangeNames[1]}`;
+      }
+      if (rangeNames.length > 2) {
+        annotationObject.name += ' &...';
+      }
+    } else {
+      annotationObject.name = 'empty';
+    }
+    annotationObject.name += ' (range)';
+  } else if (annotationObject.selection.type === 'partition') {
+    annotationObject.name = `${annotationObject.selection.partition.variable} (partition)`;
+  } else {
+    annotationObject.name = 'unknown';
```

Comment:
Both of these are simply paranoia - I don't think either can actually happen in the current interface. 
We do have the possibility of creating 'rule' selections, but they aren't actually created in any higher-level tool.